SignalServiceKit/src/Util/DataSource.m

```diff
@@ -487,12 +487,16 @@ - (BOOL)moveToUrlAndConsume:(NSURL *)dstUrl error:(NSError **)error;
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wblock-capture-autoreleasing"
                                self.isConsumed = YES;
-                               if ([[NSFileManager defaultManager] isWritableFileAtPath:self.fileUrl.path]) {
+                               // We generally prefer moving the file when the file is writable (thus removable) because
+                               // it is faster. But we should not move a file when the file is not in our container.
+                               // So we check if the file's path is under our temporary directory instead of checking
+                               // if the file is writable to decide whether we are going to move the file or copy the file.
+                               if ([self.fileUrl.path hasPrefix:OWSTemporaryDirectory()]) {
```

Comment:
I think we'll need to check if it's in our container too to be comprehensive, such as OWSFileSystem.appSharedDataDirectoryPath()